cmake/Intel.cmake

```diff
@@ -20,7 +20,8 @@ endif()
 
 if(DEBUG)
     add_definitions(-DDEBUG)
-    set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv -init=snan,arrays")
+    #set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv -init=snan,arrays")
```

Comment:
@climbfuji May I ask why we remove the "-check nopointer" option? Can we check if there are warning messages in the DEBUG compile log file? If I remember correctly, some diagnostic field computation in dycore could throw warning messages in the setting of parallel configuration even though they are not required to output.

Reply:
I can revert this, it's not needed for the PR. But it's a nice thing that we can turn the pointer checking on without the code crashing - so I would consider this as a plus.